packages/block-library/src/button/style.scss

```diff
@@ -41,15 +41,19 @@ $blocks-button__height: 56px;
 
 
 // the first selector is required for old buttons markup
-
 .wp-block-button.no-border-radius,
 .wp-block-button__link.no-border-radius {
 	border-radius: 0 !important;
 }
 
-.wp-block-button.is-style-outline .wp-block-button__link,
+.wp-block-button.is-style-outline,
 .wp-block-button__link.is-style-outline {
 	color: $dark-gray-700;
 	background-color: transparent;
+
+}
+
+.wp-block-button__link.is-style-outline,
+.wp-block-button.is-style-outline .wp-block-button__link {
```

Comment:
This approach uses the `.wp-block-button` while maintains the 020 specificity for the color related properties (the same as WP 5.4).

However, it needs 030 to target the border for the link element, which is an increment from WP 5.4's 020. I've checked some common themes (defaults + popular vendors) and they use 020 or 030 within the same stylesheet. In practice, I haven't seen any theme setting different border styles, although I can't check all and many I reviewed do add some code for borders (interestingly, the same than core's in WP 5.4):

```
.is-style-outline .wp-block-button__link {
	border: 2px solid;
}
```

To maintain backward-compatibility I see two ways forward:

1. Revert to what we had in WP 5.4 (and before?): `.is-style-outline .wp-block-button__link`.
2. Use `.wp-block-button.is-style-outline` for color-related properties and `.is-style-outline .wp-block-button__link` for the border.

I don't have a strong preference, perhaps I'd be inclined to go with 1 for simplicity, given that's what people is used to and would probably have its workarounds (like, not adding an `outline` style in a block that can be a parent of a button).

Thoughts @youknowriad?

Reply:
The approach is good IMO but I think we're targetting different things for the colors right? Is this ok?

Previous:

`.is-style-outline .wp-block-button__link,` this targets the link classi inside outlined buttons.

Now:

`.wp-block-button.is-style-outline`  this  targets the parent element of the link class?

Reply:
Ah, good catch, you're absolutely right: we need to set the link to transparent instead of setting the parent. I missed it in my testing because themes also tend to set the background to transparent.

Pushed the changes to revert to what we had in 5.4.